### PR TITLE
Support more jobs for the prison

### PR DESCRIPTION
**Pull request by bicatcho11**
## Description

Support more jobs for the prison

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: client/jobs.lua [hunk 1]
```diff
@@ -34,25 +33,23 @@ function CreateJobBlip() -- Used globally
...
-    if currentBox == #electricalBoxes then
-        electricalBoxes = newRandomArray(#Config.Jobs.electrician.locations)
-        currentBox = 1
+    if currentTask == #jobLocations then
+        jobLocations = newRandomArray(#Config.Jobs[prisonJob].locations)
+        currentTask = 1
         TriggerServerEvent('qbx_prison:server:completedJob')
     else
-        currentBox += 1
+        currentTask = currentTask + 1
```
Comment (BerkieBb) [inline client/jobs.lua]: Use += here

File: client/jobs.lua [hunk 2]
```diff
@@ -81,46 +78,58 @@ local function startWork()
...
-                    end
-                end,
-            })
+                })
+            else
+                lib.zones.box({
+                    coords = coords,
+                    size = vec3(3, 5, 3),
+                    onEnter = function()
+                        lib.showTextUI(locale("info.job_interaction"))
```
Comment (BerkieBb) [inline client/jobs.lua]: Don't forget to change the locales

File: client/jobs.lua [hunk 3]
```diff
@@ -81,46 +78,58 @@ local function startWork()
     StopAnimTask(cache.ped, "anim@gangops@facility@servers@", "hotwire", 1.0)
 end
 
-local function canInteractWithBox(i)
-    return InJail and CurrentJob and not isWorking and i == currentBox
+local function canInteractWithTask(i)
+    return InJail and prisonJob and not isWorking and i == currentTask
+end
+
+function chooseJob(job)
```
Comment (BerkieBb) [inline client/jobs.lua:85]: Make this a local function

File: client/jobs.lua [hunk 4]
```diff
@@ -81,46 +78,58 @@ local function startWork()
...
+    prisonJob = job
+    jobLocations = newRandomArray(#Config.Jobs[prisonJob].locations)
 end
 
--- Threads
+function getPrisonJob()
```
Comment (BerkieBb) [inline client/jobs.lua]: This can just be directly passed as a function to the export, no need to create a new variable for it

**Review by BerkieBb — CHANGES_REQUESTED** [review]

File: client/jobs.lua [hunk 5]
```diff
@@ -81,46 +78,56 @@ local function startWork()
...
+    return InJail and prisonJob and not isWorking and i == currentTask
+end
+
+function chooseJob(job)
+    prisonJob = job
+    jobLocations = newRandomArray(#Config.Jobs[prisonJob].locations)
 end
 
--- Threads
+exports('getPrisonJob', function()
```
Comment (Manason) [inline client/jobs.lua:90]: What will this export be used for?

File: client/jobs.lua [hunk 6]
```diff
@@ -13,16 +16,12 @@ local function newRandomArray(size)
     return array
 end
 
-local electricalBoxes = newRandomArray(#Config.Jobs.electrician.locations)
-local currentBox = 1
-
---- This will create the blip for the current prison job
-function CreateJobBlip() -- Used globally
+local function CreateJobBlip()
```
Comment (Manason) [inline client/jobs.lua]: camelCase local function names

File: client/jobs.lua [hunk 7]
```diff
@@ -1,5 +1,8 @@
-CurrentBlip = 0
+local prisonJob = nil
+local CurrentBlip = 0
```
Comment (Manason) [inline client/jobs.lua]: camelCase local variable names

File: client/jobs.lua [hunk 8]
```diff
@@ -81,46 +78,56 @@ local function startWork()
...
+    if prisonJob then
+        for i = 1, #Config.Jobs[prisonJob].locations do
+            local coords = Config.Jobs[prisonJob].locations[i]
+            if Config.UseTarget then
+                exports.ox_target:addBoxZone({
+                    coords = coords,
+                    size = vec3(1.5, 1.6, 5),
+                    options = {
+                        {
+                            icon = 'fa-solid fa-bolt',
```
Comment (Manason) [inline client/jobs.lua]: Seems like you'd want the icon to change based on the job, no?

File: client/jobs.lua [hunk 9]
```diff
@@ -81,46 +78,56 @@ local function startWork()
...
+                            label = locale("info.job_interaction_target", prisonJob),
```
Comment (Manason) [inline client/jobs.lua]:
```suggestion
                            label = locale("info.job_interaction_target", prisonJob.label),
```
Does this locale need to change as well?

File: client/jobs.lua [hunk 10]
```diff
@@ -34,25 +33,23 @@ function CreateJobBlip() -- Used globally
...
     else
-        currentBox += 1
+        currentTask += 1
     end
 
     CreateJobBlip()
 end
 
---- This will be triggered once you interact with a job location to perform your job at
 local function startWork()
```
Comment (Manason) [inline client/jobs.lua:53]: You'd want to change the animation and locales of the progress bar depending on the job, no?

**Review by Manason — CHANGES_REQUESTED** [review]

File: client/jobs.lua (hunk 5 above)
Comment (bicatcho11) [reply client/jobs.lua:90]: In case of using things other than the target

**Comment by BerkieBb** [comment]
Looks good now, just need to solve conflicts and have you tested it yourself?

**Pull request closed without merging**